velox/functions/prestosql/Repeat.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+class RepeatFunction : public exec::VectorFunction {
+ public:
+  bool isDefaultNullBehavior() const override {
+    // For repeat(null, count), it should output an array with
+    // repeating null for "count" times.
+    return false;
+  }
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    VELOX_CHECK_EQ(args.size(), 2);
+
+    const auto& elementVector = args[0].get();
+    auto countVector = args[1]->asFlatVector<int32_t>();
```

Comment:
Inputs are not guaranteed to be flat. Use DecodedArgs.

Reply:
Thanks for pointing this out. I think this is the cause of the fuzzer crashes. It could be Constant Vector which is not flat. Given that the function signature is `repeat(T, int)`, is this the only case that the args[1] is not flat? If so, is args[1]->as<SimpleVector<int32_t>>() good enough, or do we have to use `DecodedArgs`?

Reply:
Is the second argument always const or can be an expression?
We could also write an optimized pass for the const vector (majority of use cases) if that makes sense.

Reply:
@zacw7 Zac, I suggest you reach out to @laithsakka (Laith) or @kagamiori (Wei) to get an overview of expression evaluation and functions and understand the possible combinations of encodings.